source/mainnet/smart-contracts/guides/unit-test-contract.rst

```diff
@@ -1,9 +1,19 @@
+:orphan:
+
 .. _unit-test-contract:
 
 ============================
 Unit test a contract in Rust
 ============================
 
+.. warning::
+
+   Unit testing your contracts with the |test_infrastructure|_ has been deprecated in favor of |concordium-smart-contract-testing|_.
+   To migrate your contracts and tests see :ref:`migrate-contracts-for-std-9` and :ref:`integration-test-contract`.
+
+   If you are not ready to migrate your contracts, you can still use the guide below.
+   Just make sure to add ``#[allow(deprecated)]`` attribute above your test module to avoid warnings.
```

Comment:
Suggestion: It could be a code snippet added here to give an example of where to add the ``#[allow(deprecated)]`` on the test module.